src/rulesets/Base/fastmath_able.jl

```diff
@@ -36,10 +36,94 @@ let
 
 
         # Unary complex functions
-        @scalar_rule abs(x::Real) sign(x)
-        @scalar_rule abs2(x) 2x
-        @scalar_rule angle(x::Real) Zero()
-        @scalar_rule conj(x::Real) One()
+        ## abs
+        function frule((_, Δx), ::typeof(abs), x::Real)
+            return abs(x), sign(x) * real(Δx)
+        end
+        function frule((_, Δz), ::typeof(abs), z::Complex)
+            Ω = abs(z)
+            return Ω, (real(z) * real(Δz) + imag(z) * imag(Δz)) / ifelse(iszero(z), one(Ω), Ω)
```

Comment:
```suggestion
            return Ω, (real(z) * real(Δz) + imag(z) * imag(Δz)) / ifelse(iszero(z), one(Ω), Ω)
            # `ifelse` is applied only to denominator to ensure type-stability. 
```
Hope I am not putting the wrong words into your mouth.